addons/website_forum/controllers/website_forum.py

```diff
@@ -366,12 +366,13 @@ def question_reopen(self, forum, question, **kwarg):
 
     @http.route('/forum/<model("forum.forum"):forum>/question/<model("forum.post"):question>/delete', type='http', auth="user", methods=['POST'], website=True)
     def question_delete(self, forum, question, **kwarg):
-        question.active = False
+        question.state = 'deleted'
```

Comment:
there can still be made archived records by internal users, we just don't want moderators to create those records from front-end. And I'm not sure how we would differentiate the two in older versions if we were to write a migration script...

Reply:
IMO it something is "deleted" from a functionnal POV, it should be after your change as well

Reply:
> there can still be made archived records by internal users

so it will be marked as deleted in the frontend right ? (so it's state should be deleted now)

Reply:
So the idea is that: in the back-end user should be able to archive the question, in which case it will not change the state but have active field false. In front-end user will be able to change the state from active to offensive to deleted, but the active field will always stay True.

Reply:
Yes ok, that'sq how it work now

But before your change, if a post is archived, it's marked as deleted

Now it won't be displayed as deleted on the frontend and it should

Reply:
So you should do a migration script to keep the functionnal state of the post the same

Reply:
Just do the test, make a db in master, make a screenshot of the frontend (of a deleted post), then upgrade

It should be the exact same after the upgrade

Reply:
I think it's part of the changes we want here.
> In front-end user will be able to change the state from active to offensive to deleted, but the active field will always stay True.

this is not how it works now, but how it should.
An archived post shouldn't show at all on the front-end.

Reply:
@flch-odoo So it's ok if someone delete a post in 16.0 (so it's archived), it's displayed as deleted on the frontend, and after the 17.0 upgrade is not displayed at all and the state is still the old one ?

> I think it's part of the changes we want here.

Yes I know, the point is to keep the functionnal state of the posts after the upgrade

Reply:
to discuss tomorrow at GR :)